src/Capgemini.PowerApps.PackageDeployerTemplate/Services/EnvironmentVariableDeploymentService.cs

```diff
@@ -0,0 +1,124 @@
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger"/>.</param>
+        /// <param name="crmSvc">The <see cref="ICrmServiceAdapter"/>.</param>
+        public EnvironmentVariableDeploymentService(ILogger logger, ICrmServiceAdapter crmSvc)
+        {
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this.crmSvc = crmSvc ?? throw new ArgumentNullException(nameof(crmSvc));
+        }
+
+        /// <summary>
+        /// Sets environment variables on the target Power Apps environment.
+        /// </summary>
+        /// <param name="environmentVariables">A dictionary of keys and values to set.</param>
+        public void SetEnvironmentVariables(IDictionary<string, string> environmentVariables)
+        {
+            if (environmentVariables is null || !environmentVariables.Any())
+            {
+                this.logger.LogInformation("No environment variables have been configured.");
+                return;
+            }
+
+            foreach (KeyValuePair<string, string> entry in environmentVariables)
+            {
+                this.SetEnvironmentVariable(entry.Key, entry.Value);
+            }
+        }
+
+        /// <summary>
+        /// Sets an environment variable on the target Power Apps environment.
+        /// </summary>
+        /// <param name="key">Environment variable key.</param>
+        /// <param name="value">Environment variable value.</param>
+        public void SetEnvironmentVariable(string key, string value)
+        {
+            this.logger.LogInformation($"Setting {key} environment variable to {value}.");
+
+            var definition = this.GetDefinitionByKey(key, new ColumnSet(false));
+            if (definition == null)
+            {
+                this.logger.LogError($"Environment variable {key} not found on target instance.");
+                return;
+            }
+
+            var definitionReference = definition.ToEntityReference();
+            this.logger.LogTrace($"Found environment variable on target instance: {definition.Id}");
+
+            this.UpsertEnvironmentVariableValue(value, definitionReference);
+        }
+
+        private void UpsertEnvironmentVariableValue(string value, EntityReference definitionReference)
+        {
+            var existingValue = this.GetValueByDefinitionId(definitionReference, new ColumnSet(Constants.EnvironmentVariableValue.Fields.Value));
+            if (existingValue != null)
+            {
+                existingValue[Constants.EnvironmentVariableValue.Fields.Value] = value;
+                this.crmSvc.Update(existingValue);
+            }
+            else
+            {
+                this.SetValue(value, definitionReference);
+            }
+        }
+
+        private void SetValue(string value, EntityReference definition)
+        {
+            var val = new Entity(Constants.EnvironmentVariableValue.LogicalName)
```

Comment:
Would this variable be better called `entity` as `val` is very similar in name to `value`?